### DISCUSSION TITLE
Help for batch compositing 2 png folders (new user) - W10

### DISCUSSION THREAD

**Discussion by Kayael76** [ImageMagick/ImageMagick#4057] [Help] [answered]
I searched a lot but can't seem to find a similar case. Essentially, I have 2 png folders, one with images listed in numeric order that would serve as the background, and the other with images listed in the exact same needed order that would be layered on top. All images are the exact same dimensions so it doesn't overcomplicate things.

I would like the results in a 3rd folder as shown below, retaining the same names :

![image](https://user-images.githubusercontent.com/48800201/128638553-23b4d708-a054-4e01-b9be-04a6d0cb0ab1.png)

I'm sure it's probably very simple but I can't wrap my head around the batch command. I'm on Windows 10 using IM 7.1.0-4. Thanks for the help !

**Comment by snibgo** [comment]
I can think of two methods for doing this.

(a) Write a shell script that loops through files, and calls IM once for each pair of inputs and output.

(b) Call IM once, reading all the input files, using "-layers composite", and write all the outputs.

Method (b) is simpler, but if you have too many pixels to fit into your memory at the same time, performance will be poor or it won't work at all.

So, how many pixels in total are in your input directories? A rough guess will do.

**Reply by Kayael76** [reply]
Each image is 640*480, so that's 307200 pixels for one. I currently have a lot of images to process (more than a thousand), but I don't mind processing by small batches. If it's any help, I have a Ryzen 5 5600X and 32gb of RAM.

**Comment by snibgo** [comment, marked as answer]
640\*480\*1000 is 0.3 gigapixels. This might all fit in your RAM. 100 files should be fine.

Anyhow, the basic command for method (a) two inputs and one output is probably something like:
```
magick bg\001.png layer\001.png -compose Over -composite results\001.png
```
... assuming the names of one set of files is 001.png. You should test this: does it do what you want? Then you would put that in a shell "for" loop.

For method (b), all the files in a single command:
```
magick bg\* NULL: layer\* -compose Over -layers composite results\r.png
```
That will create output files named r-0.png, r-1.png etc. If you want them named after the files in bg, then:
```
magick ^
  bg\* ^
   -set filename:MYFILE "%%t" ^
  NULL: ^
  layer\* -compose Over -layers composite results\%%[filename:MYFILE].png
```
(Windows BAT syntax.)

**Reply by Kayael76** [reply]
The second and third method worked liked a charm, it's also helping me understand the syntax. Thank you very much !
